src/main/java/uk/gov/hmcts/darts/dailylist/service/DailyListProcessorImpl.java

```diff
@@ -0,0 +1,181 @@
+    private final DefenceRepository defenceRepository;
+    private final ProsecutorRepository prosecutorRepository;
+
+    private final ObjectMapper objectMapper;
+
+    @Override
+    public void processAllDailyLists(LocalDate date) {
+        Arrays.stream(SourceType.values()).forEach(sourceType -> processDailyListForSourceType(date, sourceType));
+    }
+
+    private void processDailyListForSourceType(LocalDate date, SourceType sourceType) {
+        List<CourthouseEntity> allCourthouses = courthouseRepository.findAll();
+        for (CourthouseEntity allCourthouse : allCourthouses) {
+            List<DailyListEntity> dailyLists = dailyListRepository.findByCourthouse_IdAndStatusAndStartDateAndSourceOrderByPublishedTimestampDesc(
+                    allCourthouse.getId(),
+                    String.valueOf(JobStatusType.NEW),
+                    date, String.valueOf(sourceType)
+            );
+
+            // Daily lists are being ordered descending by date so first item will be the most recent version
+            if (!dailyLists.isEmpty()) {
+                try {
+                    processDailyList(dailyLists.get(0));
+                } catch (JsonProcessingException e) {
+                    throw new DartsApiException(DailyListError.FAILED_TO_PROCESS_DAILYLIST, e);
+                }
+
+            }
+            if (dailyLists.size() > 1) {
+                ignoreOldDailyList(dailyLists.subList(1, dailyLists.size()));
+            }
+        }
+    }
+
+    private void ignoreOldDailyList(List<DailyListEntity> dailyLists) {
+        for (DailyListEntity dailyList : dailyLists) {
+            dailyList.setStatus(String.valueOf(JobStatusType.IGNORED));
+        }
+    }
+
+    @SuppressWarnings("checkstyle:VariableDeclarationUsageDistance")
+    private void processDailyList(DailyListEntity dailyListEntity) throws JsonProcessingException {
+        DailyList dailyList = objectMapper.readValue(dailyListEntity.getContent(), DailyList.class);
+        JobStatusType statusType = JobStatusType.PROCESSED;
+
+        for (CourtList courtList : dailyList.getCourtLists()) {
+
+            String courtHouseName = courtList.getCourtHouse().getCourtHouseName();
+            Optional<CourthouseEntity> foundCourthouse = courthouseRepository.findByCourthouseNameIgnoreCase(
+                    courtHouseName);
+
+            if (foundCourthouse.isPresent()) {
+                List<Sitting> sittings = courtList.getSittings();
+                for (Sitting sitting : sittings) {
+                    List<Hearing> hearings = sitting.getHearings();
+                    for (Hearing dailyListHearing : hearings) {
+
+                        String caseNumber = getCaseNumber(dailyListHearing.getDefendants(), dailyListEntity, dailyListHearing);
+
+                        HearingEntity hearing = retrieveCoreObjectService.retrieveOrCreateHearing(
+                                courtHouseName, String.valueOf(sitting.getCourtRoomNumber()),
```

Comment:
General thought, not related to your code, but wondering if the retrieveCoreObjectService methods should be trimming the courtroom number and case number etc?

Reply:
That is a good idea but I think all sanitising inputs should be done before saving into the database. I guess daily lists are an exception as we get it as a massive string.